src/storage/MonitoringStore.ts

```diff
@@ -29,31 +30,38 @@ export class MonitoringStore<T extends ResourceStore = ResourceStore>
   }
 
   public async addResource(container: ResourceIdentifier, representation: Representation,
-    conditions?: Conditions): Promise<ResourceIdentifier> {
+    conditions?: Conditions): Promise<ModifiedResource> {
     const identifier = await this.source.addResource(container, representation, conditions);
-    this.emitChanged([ container, identifier ]);
+    this.emitChanged([ changedResource(container), identifier ]);
     return identifier;
   }
 
   public async deleteResource(identifier: ResourceIdentifier,
-    conditions?: Conditions): Promise<ResourceIdentifier[]> {
+    conditions?: Conditions): Promise<ModifiedResource[]> {
     return this.emitChanged(await this.source.deleteResource(identifier, conditions));
   }
 
   public async setRepresentation(identifier: ResourceIdentifier, representation: Representation,
-    conditions?: Conditions): Promise<ResourceIdentifier[]> {
+    conditions?: Conditions): Promise<ModifiedResource[]> {
     return this.emitChanged(await this.source.setRepresentation(identifier, representation, conditions));
   }
 
   public async modifyResource(identifier: ResourceIdentifier, patch: Patch,
-    conditions?: Conditions): Promise<ResourceIdentifier[]> {
+    conditions?: Conditions): Promise<ModifiedResource[]> {
     return this.emitChanged(await this.source.modifyResource(identifier, patch, conditions));
   }
 
-  private emitChanged(identifiers: ResourceIdentifier[]): typeof identifiers {
-    for (const identifier of identifiers) {
-      this.emit('changed', identifier);
+  private emitChanged(modified: ModifiedResource[]): ModifiedResource[] {
+    // Don't emit 'changed' event for internal resources
+    if (!this.isInternalResource(modified)) {
+      this.emit('changed', modified);
     }
-    return identifiers;
+    return modified;
+  }
+
+  private isInternalResource(result: ModifiedResource[]): boolean {
+    return result.filter(
```

Comment:
Also better done with a `.some`, so you don't have to collect them all and count.

Reply:
Now using `.some` instead of `.filter` (https://github.com/solid/community-server/pull/1075/commits/461a9361621353f3af3b65d09dddb2d633e96efa)

*This `isInternalResource` function should probably be deleted anyway since it has been recommended in another review comments that this class should not filter out `.internal` event emissions anyway.*